apps/backend/src/user/user.service.ts

```diff
@@ -0,0 +1,93 @@
+import { Injectable } from '@nestjs/common';
+import { UserEntity } from './entities/user.entity';
+import { InjectRepository } from '@mikro-orm/nestjs';
+import { UserDto, UserPublicDTO } from '@datatlas/shared/models';
+import { EntityRepository } from '@mikro-orm/core';
+
+@Injectable()
+export class UserService {
+  constructor(
+    @InjectRepository(UserEntity)
+    private readonly userRepository: EntityRepository<UserEntity>
+  ) {}
+
+  /**
+   * Adds a user into the database.
+   * TODO : passwords are stored clearly in database -> hash them !
+   * @param userDto Private user data.
+   * @return number New user_id or 0 if user already exists.
+   */
+  async createUser(userDto: UserDto): Promise<number> {
+    const user = new UserEntity(userDto.username, userDto.password, userDto.role, userDto.active);
+    return (await this.isUsernameAlreadyInDatabase(user.username))
+      ? 0
+      : this.userRepository.persistAndFlush(user).then(() => {
+          return this.getUserIDByUserName(user.username);
+        });
+  }
+
+  /**
+   * Get public user data from its user_id
+   * @param id
+   * @return UserPublicDTO public user data {id, username, role, is_active}
+   */
+  async getUser(id = 0): Promise<UserPublicDTO> {
+    return this.userRepository
+      .findOne({ id })
+      .then((data_user) => new UserPublicDTO(data_user.id, data_user.username, data_user.role, data_user.active));
+  }
+
+  async updateUser(user: { userId: number } & UserDto): Promise<void> {
+    /*
+      How to proceed.
+      -> Create new UserEntity with old data already stored (get them with the user id given in args).
+      -> Update this object with new data given in args.
+      -> Flush.
+     */
+    const id = user.userId;
+    return this.userRepository.findOne({ id }).then((data_user) => {
+      data_user.username = user.username;
+      data_user.password = user.password;
+      data_user.role = user.role;
+      data_user.active = user.active;
+      return this.userRepository.flush();
+    });
+  }
+
+  /**
+   * Deletes user. Returns void.
+   * @param user_id
+   * @return void
+   */
+  async deleteUserById(user_id: number): Promise<void> {
+    return this.userRepository.nativeDelete(user_id).then(() => {
```

Comment:
Case should be consistent :
```suggestion
  async deleteUserById(userId: number): Promise<void> {
    return this.userRepository.nativeDelete(userId).then(() => {
```